packages/swapper/src/swappers/ChainflipSwapper/endpoints.ts

```diff
@@ -0,0 +1,343 @@
+  UtxoFeeData,
+} from '../../types'
+import { isExecutableTradeQuote, isExecutableTradeStep, isToken } from '../../utils'
+import { CHAINFLIP_BAAS_COMMISSION, chainIdToChainflipNetwork } from './constants'
+import type { ChainflipBaasSwapDepositAddress } from './models/ChainflipBaasSwapDepositAddress'
+import { getTradeQuote, getTradeRate } from './swapperApi/getTradeQuote'
+import type { ChainFlipStatus } from './types'
+import { chainflipService } from './utils/chainflipService'
+import { getLatestChainflipStatusMessage } from './utils/getLatestChainflipStatusMessage'
+
+// Persists the ID so we can look it up later when checking the status
+const tradeQuoteMetadata: Map<string, ChainflipBaasSwapDepositAddress> = new Map()
+
+export const chainflipApi: SwapperApi = {
+  getTradeQuote,
+  getTradeRate,
+  getUnsignedEvmTransaction: async ({
+    chainId,
+    from,
+    tradeQuote,
+    assertGetEvmChainAdapter,
+    config,
+    supportsEIP1559,
+  }: GetUnsignedEvmTransactionArgs): Promise<EvmTransactionRequest> => {
+    if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
+
+    const brokerUrl = config.REACT_APP_CHAINFLIP_API_URL
+    const apiKey = config.REACT_APP_CHAINFLIP_API_KEY
+
+    const step = tradeQuote.steps[0]
+    const sellChainflipChainKey = `${step.sellAsset.symbol.toLowerCase()}.${
+      chainIdToChainflipNetwork[step.sellAsset.chainId]
+    }`
+    const buyChainflipChainKey = `${step.buyAsset.symbol.toLowerCase()}.${
+      chainIdToChainflipNetwork[step.buyAsset.chainId]
+    }`
+
+    // Subtract the BaaS fee to end up at the final displayed commissionBps
+    let serviceCommission = parseInt(tradeQuote.affiliateBps) - CHAINFLIP_BAAS_COMMISSION
+    if (serviceCommission < 0) serviceCommission = 0
+
+    const maybeSwapResponse = await chainflipService.get<ChainflipBaasSwapDepositAddress>(
+      `${brokerUrl}/swap` +
+        `?apiKey=${apiKey}` +
+        `&sourceAsset=${sellChainflipChainKey}` +
+        `&destinationAsset=${buyChainflipChainKey}` +
+        `&destinationAddress=${tradeQuote.receiveAddress}` +
+        `&boostFee=10` +
+        // TODO: Calculate minprice based on tradeQuote.slippageTolerancePercentageDecimal, step.sellAmountIncludingProtocolFeesCryptoBaseUnit, step.buyAmountAfterFeesCryptoBaseUnit
+        // `&minimumPrice=` +
+        // `&refundAddress=${from}` +
+        // `&retryDurationInBlocks=10` +
+        `&commissionBps=${serviceCommission}`,
+
+      // TODO: Below is the reference code of Chainflip to calculate the minPrice parameter
+      //     const tolerance = new BigNumber(params.slippageTolerancePercent);
+      //     const estimatedPrice = new BigNumber(quote.estimatedPrice);
+      //     minPrice = estimatedPrice
+      //       .times(new BigNumber(100).minus(tolerance).dividedBy(100))
+      //       .toFixed(assetConstants[destAsset].decimals);
+
+      // TODO: For DCA swaps we need to add the numberOfChunks/chunkIntervalBlocks parameters
```

Comment:
Not for this PR, streaming swaps have a lot of ties to THORChain and will require a separate unit of work to decouple from it